test/suite/test_hs21.py

```diff
@@ -0,0 +1,181 @@
+        config = meta_cursor[uri]
+        meta_cursor.close()
+        # The search string will look like: 'run_write_gen=<num>'.
+        # Just reverse the string and take the digits from the back until we hit '='.
+        write_gen = re.search('run_write_gen=\d+', config)
+        self.assertTrue(write_gen is not None)
+        write_gen_str = str()
+        for c in reversed(write_gen.group(0)):
+            if not c.isdigit():
+                self.assertEqual(c, '=')
+                break
+            write_gen_str = c + write_gen_str
+        return int(write_gen_str)
+
+    def test_hs(self):
+        active_files = []
+        value1 = 'a' * 500
+        value2 = 'd' * 500
+
+        # Set up 'numfiles' with 'numrows' entries. We want to create a number of files that
+        # contain active history (content newer than the oldest timestamp).
+        for f in range(self.numfiles):
+            table_uri = 'table:%s.%d' % (self.file_name, f)
+            file_uri = 'file:%s.%d.wt' % (self.file_name, f)
+            # Create a small table.
+            ds = SimpleDataSet(
+                self, table_uri, 0, key_format='S', value_format='S', config='log=(enabled=false)')
+            ds.populate()
+            # Checkpoint to ensure we write the files metadata checkpoint value.
+            self.session.checkpoint()
+            # Get the base write gen of the file so we can compare after the handles get closed.
+            base_write_gen = self.parse_run_write_gen(file_uri)
+            active_files.append((base_write_gen, ds))
+
+        # Pin oldest and stable to timestamp 1.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(1) +
+            ',stable_timestamp=' + timestamp_str(1))
+
+        # Perform a series of updates over our files to check the history store is
+        # working with old and new timestamps.
+        for (_, ds) in active_files:
+            # Load data.
+            self.large_updates(ds.uri, value1, ds, self.nrows // 2 , 1)
+            # Check that all updates are seen.
+            self.check(value1, ds.uri, self.nrows // 2, 1)
+
+            # Load more data with a later timestamp.
+            self.large_updates(ds.uri, value2, ds, self.nrows, 100)
+            # Check that the new updates are only seen after the update timestamp.
+            self.check(value1, ds.uri, self.nrows // 2, 1)
+            self.check(value2, ds.uri, self.nrows, 100)
+
+        # Our sweep scan interval is every 1 second and the amount of idle time needed for a handle to be closed is 3 seconds.
+        # It should take roughly 4 seconds for the sweep server to close our file handles. Lets wait at least double
+        # that to be safe.
+        max = 8
+        sleep = 0
+        # After waiting for the sweep server to remove our idle handles, the only open
+        # handles that should be the metadata file, history store file and lock file.
+        final_numfiles = 3
+        while sleep < max:
```

Comment:
I am curious on average how long this test takes to execute. We have to be conscious of not adding a lot to the runtime, else PR testing starts to take longer.

Reply:
The test takes on average ~10 seconds to complete (the loop in this comment is ~4-5 seconds). 
Should I shorten the `close_idle_time` time? (I was mostly trying to pick an interval of time that could be consistently used to observe the closing of the handles)

Reply:
I want to say yes to reducing the `close_idle_time` to 2, or even 1 sec. If we protect the test from not failing if the handle closes sooner than expected, we should be fine. Do you see any issues with reducing the closing idle time?

Maybe we should run the test a few times with different close_idle_time and see what happens.

Reply:
Updated it with a `close_idle_time` of 2. 
I experimented with 1 and 2 seconds, though in the end it had little effect on the overall test run time between the two configurations.

Reply:
Cool, what is the average run time of the test now ?

Reply:
Roughly ~8 seconds. Most of the test time at this point is around the update and checking logic.